api/storage/v1alpha1/s3bucket_types.go

```diff
@@ -0,0 +1,121 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	runtimev1alpha1 "github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// S3BucketParameters define the desired state of a Cloudscale S3 Bucket
+// https://www.cloudscale.ch/en/api/v1#objects-users
+// https://docs.ceph.com/docs/bobtail/radosgw/s3/bucketops/
+type S3BucketParameters struct {
+	// Tags are optional key, value pairs to add to an S3 bucket
+	// +optional
+	Tags *map[string]string `json:"tags,omitempty"`
+
+	// CannedACL applies a built-in ACL for common bucket use cases.
+	// +kubebuilder:validation:Enum=private;public-read;public-read-write;authenticated-read
+	// +optional
+	CannedACL *string `json:"cannedACL,omitempty"`
+
+	// Region of the bucket.
+	// +kubebuilder:validation:Enum=lpg;rma
+	Region string `json:"region"`
+}
+
+// S3BucketSpec defines the desired state of S3Bucket
+type S3BucketSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  S3BucketParameters `json:"forProvider,omitempty"`
+}
+
+// S3BucketObservation is the representation of the current state that is observed.
+type S3BucketObservation struct {
+	Status       string `json:"status,omitempty"`
+	ObjectUserID string `json:"objectUserId,omitempty"`
+}
+
+// S3BucketStatus defines the observed state of S3Bucket
+type S3BucketStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+
+	AtProvider S3BucketObservation `json:"atProvider,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// S3Bucket is the Schema for the s3buckets API
+// +kubebuilder:printcolumn:name="CLASS",type="string",JSONPath=".spec.classRef.name"
+// +kubebuilder:printcolumn:name="STATUS",type="string",JSONPath=".status.atProvider.status"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:resource:scope=Namespaced
```

Comment:
This one should be cluster scoped, the same as `S3BucketClass`. As of Crossplane 0.4 only resource claims are namespaced.